stubs/PyAutoGUI/pyautogui/__init__.pyi

```diff
@@ -1,37 +1,177 @@
 import contextlib
+from _typeshed import Incomplete
 from collections.abc import Callable, Generator, Iterable, Sequence
 from datetime import datetime
-from typing import NamedTuple, TypeVar, overload
-from typing_extensions import ParamSpec
+from typing import NamedTuple, SupportsFloat, SupportsInt, TypeVar, overload
+from typing_extensions import ParamSpec, SupportsIndex, TypeAlias
 
-# from pyscreeze import Box
+from PIL import Image
 
 class PyAutoGUIException(Exception): ...
 class FailSafeException(PyAutoGUIException): ...
 class ImageNotFoundException(PyAutoGUIException): ...
 
 _P = ParamSpec("_P")
 _R = TypeVar("_R")
+_NormalizeableXArg: TypeAlias = SupportsInt | Sequence[SupportsInt]
+_Unused: TypeAlias = object
 
-# TODO: Complete types with pyscreeze once we can import it as a type dependency
-# Actually `pyscreeze.Box`, but typeshed doesn't currently have stubs for pyscreeze
-# (and the library doesn't have type annotations either)
+# TODO: cv2.Mat is not available as a type yet: https://github.com/microsoft/python-type-stubs/issues/211
+# cv2.Mat is just an alias for a numpy NDArray, but can't import that either.
+_Mat: TypeAlias = Incomplete
 
-class _Box(NamedTuple):
+class _Box(NamedTuple):  # Same as pyscreeze.Box
     left: int
     top: int
     width: int
     height: int
 
 def raisePyAutoGUIImageNotFoundException(wrappedFunction: Callable[_P, _R]) -> Callable[_P, _R]: ...
 
-# These functions reuse pyscreeze functions directly. See above TODO.
-def locate(*args, **kwargs) -> _Box | None: ...
-def locateAll(*args, **kwargs) -> Generator[_Box, None, None]: ...
-def locateAllOnScreen(*args, **kwargs) -> Generator[_Box, None, None]: ...
-def locateCenterOnScreen(*args, **kwargs) -> Point | None: ...
-def locateOnScreen(*args, **kwargs) -> _Box | None: ...
-def locateOnWindow(*args, **kwargs) -> _Box | None: ...
+# These functions reuse pyscreeze functions directly.
+# TODO: Once pyscreeze is typed, we can alias them to simplify this stub
+
+# _locateAll_opencv
+@overload
+def locate(
+    needleImage: str | _Mat,
+    haystackImage: str | _Mat,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: SupportsFloat = ...,
+) -> _Box | None: ...
+
+# _locateAll_python / _locateAll_pillow
+@overload
+def locate(
+    needleImage: str | Image.Image,
+    haystackImage: str | Image.Image,
+    grayscale: bool | None = ...,
+    limit: _Unused = ...,
+    region: _Box | None = ...,
+    step: int = ...,
+    confidence: None = ...,
+) -> _Box | None: ...
+
+# _locateAll_opencv
+@overload
+def locateAll(
+    needleImage: str | _Mat,
+    haystackImage: str | _Mat,
```

Comment:
These can both be `PIL` images -- `_locateAll_opencv` passes the arguments to `_load_cv2`: https://github.com/asweigart/pyscreeze/blob/b693ca9b2c964988a7e924a52f73e15db38511a8/pyscreeze/__init__.py#L201-L203

And `_load_cv2` explicitly accounts for the possibility that these might be `PIL` images: https://github.com/asweigart/pyscreeze/blob/b693ca9b2c964988a7e924a52f73e15db38511a8/pyscreeze/__init__.py#L173-L174

The same point applies for all the other overloaded functions that pass arguments to `locateAll()`

```suggestion
    needleImage: str | Image.Image | _Mat,
    haystackImage: str | Image.Image | _Mat,
```

Reply:
Totally missed that thanks! So it applies to all the opencv overloads since they all end up calling locateAll (even locate)